packages/gatsby-remark-prismjs/src/__tests__/index.js

```diff
@@ -100,5 +100,28 @@ describe(`remark prism plugin`, () => {
       plugin({ markdownAST })
       expect(markdownAST).toMatchSnapshot()
     })
+    it(`should not wrap keywords with <span class="token extensionTokenName"> if no extension given`, () => {
+      const code = `\`\`\`c\naRandomTypeKeyword var = 32\n\``
+      const markdownAST = remark.parse(code)
+
+      plugin({ markdownAST })
+      expect(markdownAST).toMatchSnapshot()
+    })
+    it(`should wrap keywords with <span class="token extensionTokenName"> based on given extension`, () => {
+      const code = `\`\`\`c\naRandomTypeKeyword var = 32\n\``
+      const markdownAST = remark.parse(code)
+
+      const config = {
+        languageExtensions: {
+          extend: `c`,
+          definition: {
+            extended_keywords: /(aRandomTypeKeyword)/,
+          },
+        },
+      }
+
+      plugin({ markdownAST }, config)
+      expect(markdownAST).toMatchSnapshot()
```

Comment:
Could we test something specifically (e.g. the AST) rather than relying on a snapshot to test something here?

In other words--a snapshot is valuable to ensure there aren't any regressions in the general look of the AST--but it shouldn't necessarily be used to validate something _specific_ that can be better done with a simple check/traversal of the AST.